src/dom/Integer.ts

```diff
@@ -96,17 +96,33 @@ export class Integer extends Value(Number, IonTypes.INT, _fromJsConstructor) {
   ): boolean {
     let isSupportedType: boolean = false;
     let valueToCompare: any = null;
+    if (other instanceof Integer) {
+      isSupportedType = true;
+      if (this._bigIntValue == null && other._bigIntValue == null) {
+        valueToCompare = other.numberValue();
+      } else {
+        // One of them is a JSBI
+        valueToCompare = other.bigIntValue();
+      }
+    }
+
     if (options.onlyCompareIon) {
       // `compareOnlyIon` requires that the provided value be an ion.dom.Integer instance.
-      if (other instanceof Integer) {
-        isSupportedType = true;
+      if (this._bigIntValue == null && other._bigIntValue == null) {
+        valueToCompare = other.numberValue();
+      } else {
+        // One of them is a JSBI
```

Comment:
Sorry, I meant for my example code to be an update to the existing code and not an additional block. Some of the logic is now being repeated (unless I'm missing something?).